kratos/geometries/line_3d_2.h

```diff
@@ -403,6 +403,26 @@ class Line3D2 : public Geometry<TPointType>
     }
 
 
+    ///@}
+    ///@name Intersections
+    ///@{
+
+    /**
+     * @brief Test if this geometry intersects with other geometry
+     * @param  ThisGeometry Geometry to intersect with
+     * @return              True if the geometries intersect, False in any other case.
+     */
+    bool HasIntersection(const BaseType& rThisGeometry) const override {
+        const BaseType& geom = *this;
+        if (rThisGeometry.WorkingSpaceDimension() > geom.WorkingSpaceDimension()) {
```

Comment:
this is not exactly that it is a line

Reply:
Ok, I understand, you relay the check to the other object, which may not implement the method. Should we invert the order of the checks, check first that it is a line, otherwise call the inversed method?

Reply:
I think inverting the order of the checks does not matter.

Reply:
To follow some convention and to avoid code duplication, the geometry with higher workingspacedimension is calling the intersection algorithm.

Reply:
Maybe it is a good idea that we put a comment explaining this (that we always check from the higher working space dimension to the lower one).

Reply:
Now there is a comment in the doxygen.

Reply:
BTW, I realized that it should be Local Space Dimension (like it was in Tetrahedra)

Reply:
Perfect. Resolving conversation.